ArcGISRuntimeSDKQt_QMLSamples/Maps/DownloadPreplannedMap/DownloadPreplannedMap.qml

```diff
@@ -0,0 +1,284 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// [Legal]
+
+import QtQuick 2.6
+import QtQuick.Controls 2.12
+import QtQuick.Layouts 1.12
+import Esri.ArcGISRuntime 100.7
+import Esri.ArcGISExtras 1.1
+
+Rectangle {
+    id: rootRectangle
+    clip: true
+    width: 800
+    height: 600
+
+    readonly property url outputMapPackage: System.temporaryFolder.url + "/ArcGIS"
+    property var preplannedMapAreaList: null
+    property var preplannedArea: null
+    property var path: null
+    property var mapExists: false
+    property var viewingOnlineMaps: true
+
+    MapView {
+        id: mapView
+        anchors.fill: parent
+
+        Map {
+            id: onlineMap
+            item: portalItem
+        }
+
+        Portal {
+            id: portal
+            url: "https://www.arcgis.com"
+        }
+
+        PortalItem {
+            id: portalItem
+            portal: portal
+            itemId: "acc027394bc84c2fb04d1ed317aac674"
+
+            onLoadStatusChanged: {
+                if (loadStatus !== Enums.LoadStatusLoaded)
+                    return;
+
+                busy.visible = true;
+                offlineMapTask.load();
+            }
+        }
+
+        GraphicsOverlay {
+            id: graphicsOverlay
+            renderer: SimpleRenderer {
+                symbol: SimpleLineSymbol {
+                    style: Enums.SimpleLineSymbolStyleSolid
+                    color: "red"
+                    width: 5
+                }
+            }
+        }
+
+        OfflineMapTask {
+            id: offlineMapTask
+            onlineMap: onlineMap
+
+            onErrorChanged: console.log(error.message, error.additionalMessage);
+
+            onLoadStatusChanged: {
+                if (loadStatus !== Enums.LoadStatusLoaded)
+                    return;
+
+                offlineMapTask.preplannedMapAreas();
+                path = outputMapPackage;
+                fileFolder.url = path;
+                let worked = fileFolder.makeFolder();
+                if (!worked)
+                    console.log("Make Folder Failed");
+            }
+
+            onPreplannedMapAreasStatusChanged: {
+                if (preplannedMapAreasStatus !== Enums.TaskStatusCompleted)
+                    return;
+
+                preplannedCombo.model = offlineMapTask.preplannedMapAreaList;
+                busy.visible = false;
+
+                for (let i = 0; i < offlineMapTask.preplannedMapAreaList.count; i++) {
+                    offlineMapTask.preplannedMapAreaList.get(i).loadStatusChanged.connect(function () {
+                        if (offlineMapTask.preplannedMapAreaList.get(i).loadStatus !== Enums.LoadStatusLoaded)
+                            return;
+
+                        if (i === 0)
```

Comment:
why only for index 0? Wouldn't you want to do this for any index the user supplies?

Reply:
Ah yes this. I can remove this now. Originally I was doing this to make sure that the default selected item in the combo box would be checked to see if it already existed( They used to not be deleted upon exit). Now this is not necessary. Thanks!